src/screens/EditExercise/EditExercisePage.tsx

```diff
@@ -110,8 +110,10 @@ const EditExercisePage: React.FC<EditExercisePageProps> = ({ route }) => {
 
   useEffect(() => {
     const fetchSets = async () => {
+      setLoading(true);
       const exercise = await getExerciseInWorkout(exerciseName, workoutId);
```

Comment:
hmm, since setLoading(true) and the setLoading(false) are in the same function call, this actually doesn't rerender the function in between.  You can however, set the loading state variable to true as an initial value and once the datas are taken from the database, set it to false